src/pages/index.js

```diff
@@ -0,0 +1,299 @@
+  if (data.isLiked) {
+    likeBtn.classList.add("card__like-btn_active");
+  } else {
+    likeBtn.classList.remove("card__like-btn_active");
+  }
+  likeBtn.addEventListener("click", () => handleLikeClick(likeBtn, data._id));
+  cardElement.querySelector(".card__del-btn").addEventListener("click", () => handleDelCard(cardElement, data));
+  cardImage.addEventListener("click", () => openPreviewModal(data));
+  return cardElement;
+}
+
+function renderCard(data, container) {
+  const cardElement = getCardElement(data);
+  container.prepend(cardElement);
+}
+
+function openPreviewModal(data) {
+  dom.previewImage.src = data.link;
+  dom.previewImage.alt = data.name;
+  dom.previewCaption.textContent = data.name;
+  openModal(dom.previewModal);
+}
+
+function handleDelCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(dom.deletePostModal);
+}
+
+function handleDelSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(dom.submitDeletePostBtn, "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(dom.deletePostModal);
+      selectedCard = null;
+      selectedCardId = null;
+    })
+    .catch((err) => console.error("Error deleting card:", err))
+    .finally(() => {
+      setButtonText(dom.submitDeletePostBtn, "Delete");
+      disableBtn(dom.submitDeletePostBtn, settings);
+    });
+}
+dom.submitDeletePostBtn.addEventListener("click", handleDelSubmit);
+
+function handleAvatarEditSave(evt) {
+  evt.preventDefault();
+  setButtonText(dom.avatarEditSaveBtn, "Saving...");
+  api
+    .editAvatar({ avatar: dom.avatarEditInput.value })
+    .then((user) => {
+      dom.profileImg.src = user.avatar;
+      closeModal(dom.avatarEditModal);
+      dom.avatarEditForm.reset();
+      disableBtn(dom.avatarEditSaveBtn, settings);
+    })
+    .catch((err) => console.error("Error updating avatar:", err))
+    .finally(() => {
+      setButtonText(dom.avatarEditSaveBtn, "Save");
+      disableBtn(dom.avatarEditSaveBtn, settings);
+    });
+}
+dom.avatarEditForm.addEventListener("submit", handleAvatarEditSave);
+
+addOpenModalListener(dom.avatarEditBtn, dom.avatarEditModal, () => {
+  resetValidation(
+    dom.avatarEditModal,
+    Array.from(dom.avatarEditModal.querySelectorAll(settings.inputSelector)),
+    settings
+  );
+});
+addCloseModalListener(dom.avatarCloseBtn, dom.avatarEditModal);
+
+addOpenModalListener(dom.profileEditBtn, dom.profileEditModal, () => {
+  dom.editProfileName.value = dom.profileName.textContent;
+  dom.editProfileDescription.value = dom.profileDescription.textContent;
+  resetValidation(
+    dom.editProfileForm,
+    Array.from(dom.editProfileForm.querySelectorAll(settings.inputSelector)),
+    settings
+  );
+});
+addCloseModalListener(dom.closeEditModalBtn, dom.profileEditModal);
+
+addOpenModalListener(dom.profileAddBtn, dom.profileAddModal, () => {
+  dom.profileAddForm.reset();
```

Comment:
You should ** clear ** the inputs only after **submitting** a card rather than when you open or close popups because if you unexpectedly close the popup, you will be upset if the inputs are empty when you open it once more, and you’ll have to type the data again.